core/libp2p/multi/multibase.hpp

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_MULTIBASE_HPP
+#define KAGOME_MULTIBASE_HPP
+
+#include <string>
+#include <string_view>
+#include <utility>
+
+#include "common/buffer.hpp"
+#include "common/result.hpp"
+
+namespace libp2p::multi {
+  /**
+   * Allows to distinguish between different base-encoded binaries
+   * See more: https://github.com/multiformats/multibase
+   */
+  class Multibase {
+    using ByteBuffer = kagome::common::Buffer;
+    using FactoryResult = kagome::expected::Result<ByteBuffer, std::string>;
+
+   public:
+    /**
+     * Encodings, supported by this Multibase
+     */
+    enum class Encoding : char {
+      kBase16Lower = 'f',
+      kBase16Upper = 'F',
+      kBase58 = 'Z',
+      kBase64 = 'm'
+    };
+
+    /**
+     * Encode the incoming bytes
+     * @param bytes to be encoded
+     * @param encoding - base of the desired encoding
+     * @return encoded string WITH an encoding prefix
+     */
+    virtual std::string encode(const ByteBuffer &bytes,
+                               Encoding encoding) const = 0;
+
+    /**
+     * Decode the incoming string
+     * @param string to be decoded
+     * @return bytes, if decoding was successful, error otherwise
+     */
+    virtual FactoryResult decode(std::string_view string) const = 0;
+
+    virtual ~Multibase() = 0;
+  };
+
+  inline Multibase::~Multibase() = default;
```

Comment:
why is it here?

Reply:
We need a pure virtual destructor for abstract classes, but leaving it without definition in the same base class (or at least .cpp file) leads to a compile error. And as I don't want to create `multibase.cpp` solely for the destructor, it is here, marked as `inline`, which is also necessary